### PR TITLE
Add some Windows types

### PR DESCRIPTION
**Pull request by dcristoloveanu**

### CODE REVIEW

File: tests/ctest_ut/ctestunittests.c [hunk 1]
```diff
@@ -79,5 +79,14 @@ int main()
         }
     }
 
+    {
+        size_t temp_failed_tests = 0;
+        CTEST_RUN_TEST_SUITE(WindowsTypesTests, temp_failed_tests);
+        if (temp_failed_tests != 2) // 2 expected failing tests
+        {
+            failedTests++;
+        }
```
Comment (mattdurak) [inline tests/ctest_ut/ctestunittests.c]: isn't this missing the MSVC check?

File: tests/ctest_ut/ctestunittests.c [hunk 2]
```diff
@@ -79,5 +79,14 @@ int main()
         }
     }
 
+    {
+        size_t temp_failed_tests = 0;
+        CTEST_RUN_TEST_SUITE(WindowsTypesTests, temp_failed_tests);
+        if (temp_failed_tests != 2) // 2 expected failing tests
```
Comment (mattdurak) [inline tests/ctest_ut/ctestunittests.c]: >2) // 2 [](start = 33, length = 7)

where does 2 come from? I count 4 failing tests

File: tests/ctest_ut/ctestunittests.c (hunk 1 above)
Comment (dcristoloveanu) [reply tests/ctest_ut/ctestunittests.c]: It is! Fixing it.

---
In reply to: [309833352](https://github.com/Azure/azure-ctest/pull/34#discussion_r309833352) [](ancestors = 309833352)

File: tests/ctest_ut/ctestunittests.c (hunk 2 above)
Comment (dcristoloveanu) [reply tests/ctest_ut/ctestunittests.c]: They are 5 indeed. It was 2 when there were only 2 types. But then I added the rest and now they are 5. That's why my builds were failing :-).

---
In reply to: [309833697](https://github.com/Azure/azure-ctest/pull/34#discussion_r309833697) [](ancestors = 309833697)

File: tests/ctest_ut/windows_types_tests.c [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+#include "ctest.h"
+
+CTEST_BEGIN_TEST_SUITE(WindowsTypesTests)
+
+/* ULONG */
+
+CTEST_FUNCTION(Assert_Are_Equal_ULONG_Succeeds)
+{
+    CTEST_ASSERT_ARE_EQUAL(ULONG, 1, 1, "test arg:%d", 42);
```
Comment (mattdurak) [inline tests/ctest_ut/windows_types_tests.c]: >"test arg:%d", 42); [](start = 40, length = 19)

is this 42 relevant to the test?

Comment (dcristoloveanu) [reply tests/ctest_ut/windows_types_tests.c]: Nope. Just an arg to pass to the message formatter... Can remove it, but figured that it does not really hurt.

---
In reply to: [309940053](https://github.com/Azure/azure-ctest/pull/34#discussion_r309940053) [](ancestors = 309940053)
